src/Symfony/Component/Validator/Constraints/UniqueValidator.php

```diff
@@ -69,18 +80,37 @@ private function getNormalizer(Unique $unique): callable
         return $unique->normalizer;
     }
 
-    private function reduceElementKeys(array $fields, array $element): array
+    private function reduceElementKeys(array $fields, array|object $element): array
     {
         $output = [];
         foreach ($fields as $field) {
             if (!\is_string($field)) {
                 throw new UnexpectedTypeException($field, 'string');
             }
-            if (isset($element[$field])) {
-                $output[$field] = $element[$field];
+
+            // For no BC, because PropertyAccessor require brackets for array keys
```

Comment:
For no BC, that simple case should be handled without requiring PropertyAccess

Reply:
Yes, but check if the element is an array, and the field do not contains: "[]" or ".", to call a direct access seems more difficult to just add "[]" if needed.

In case of an array, and if we use a propertyPath syntax, we can resolve multi-dimensional array/object.

Then maybe remove the bloc in 7.0, to only use property accessor syntax.

Reply:
In a first time this is what I've done: check if element is an array, then that the key do not contains any "[]." (PropertyAccessor syntax) and the key exists in the element.

In that case directly access to the array value via the key.

And add an Else statement (not array and/or PropertyPath syntax, that contains the if statement (because the actual implementation accept that the key can be missing).